sematic/plugins/storage/local_storage.py

```diff
@@ -0,0 +1,50 @@
+# Standard Library
+import logging
+import os
+
+# Sematic
+from sematic.abstract_plugin import AbstractPlugin, PluginVersion
+from sematic.config.config import get_config
+from sematic.plugins.abstract_storage import AbstractStorage, NoSuchStorageKey
+
+logger = logging.getLogger(__name__)
+
+_PLUGIN_VERSION = (0, 1, 0)
+
+
+class LocalStorage(AbstractStorage, AbstractPlugin):
+    """
+    A local storage implementation of the `AbstractStorage` interface. Values
+    are stores in the data directory of the Sematic directory, typically at
+    `~/.sematic/data`.
+    """
+
+    @staticmethod
+    def get_author() -> str:
+        return "github.com/sematic-ai"
+
+    @staticmethod
+    def get_version() -> PluginVersion:
+        return _PLUGIN_VERSION
+
+    def set(self, key: str, value: bytes):
+        logger.debug(f"{self.__class__.__name__} Setting value for key: {key}")
+
+        dir_path = os.path.split(key)[0]
+        os.makedirs(dir_path, exist_ok=True)
+
+        with open(key, "wb") as file:
+            file.write(value)
+
+    def get(self, key: str) -> bytes:
+        try:
+            with open(os.path.join(get_config().data_dir, key), "rb") as file:
+                return file.read()
+        except FileNotFoundError:
+            raise NoSuchStorageKey(self, key)
+
+    def _get_write_location(self, namespace: str, key: str) -> str:
+        return os.path.join(get_config().data_dir, namespace, key)
+
+    def _get_read_location(self, namespace: str, key: str) -> str:
```

Comment:
It appears that `_get_read_location` and `_get_write_location` return locations using different schemes. I.e. the former gives you a file path directly and the latter gives you a portable thing from which you can derive a file path. Can we make them consistent? If not, can we name them differently to indicate the difference?

Reply:
Unfortunately since this is an API defined on the abstract class, we can't rename them based on this particular implementation. That's ok.

The reason for this asymmetry is that writes are done from the Python code which has access to the filesystem, whereas reads are done from either the Python code, or Javascript code, which does not have access to the filesystem.

Reply:
then can we have the python code for the writing still receive the "portable" version, and translate it to a final filepath locally?

Reply:
bump on my previous comment here